README.md
M|ushroom
S|uper
S|aver

MSS is a protpyte software that utlizes the Raspberry Pi system to measure temp, C02 levels, humidaty,
and other useful measurements to bring farming to the mondern era.

Created by 
Brandon Faust and Christian Jensen  


contact for coding use via email: 

CHANGELOG
MSS Alpha V 0.2.1

Last Updated: 12/23/20
Status: Working

Change Log:

v0.2.1 / 02/06/2021
Status: Working
--
-Fix main_Main() format
-Timed data log session work started in start_dls
	-Currently pulls from a variable, will be changed to set from user input
v0.2 / 12.23.20
Status: Working
--
-Started main data logging portion
  --temp is displaying
-Screen clear func now works across platform
-Remove some debunk code
-In progress ->  data logging
-Now testing on physical sense hat device, all sense hat functions enabled.
  -Need to implement rounding to functions
-Updated readme file to current


v0.1 / 12.19.20
--
Status: Working
-Rewritten Menu -> Working
-First Time Setup check -> Working
-Config File Creation



Currently In Progress:
--
-First time Config
  File creation has been implemented however there are no settings that utilize the file.
-Data logging session config
  Will be built in to settings portion with FTC
-start_dls function adding main application data logging


gui.py
#! /usr/bin/env python
#  -*- coding: utf-8 -*-
#
# GUI module generated by PAGE version 4.25.1
#  in conjunction with Tcl version 8.6
#    Sep 24, 2019 08:50:04 PM EDT  platform: Windows NT

import sys

try:
    import Tkinter as tk
except ImportError:
    import tkinter as tk

try:
    import ttk
    py3 = False
except ImportError:
    import tkinter.ttk as ttk
    py3 = True

import gui_support

def vp_start_gui():
    '''Starting point when module is the main routine.'''
    global val, w, root
    root = tk.Tk()
    top = Toplevel1 (root)
    gui_support.init(root, top)
    root.mainloop()

w = None
def create_Toplevel1(root, *args, **kwargs):
    '''Starting point when module is imported by another program.'''
    global w, w_win, rt
    rt = root
    w = tk.Toplevel (root)
    top = Toplevel1 (w)
    gui_support.init(w, top, *args, **kwargs)
    return (w, top)

def destroy_Toplevel1():
    global w
    w.destroy()
    w = None

class Toplevel1:
    def __init__(self, top=None):
        '''This class configures and populates the toplevel window.
           top is the toplevel containing window.'''
        _bgcolor = '#d9d9d9'  # X11 color: 'gray85'
        _fgcolor = '#000000'  # X11 color: 'black'
        _compcolor = '#d9d9d9' # X11 color: 'gray85'
        _ana1color = '#d9d9d9' # X11 color: 'gray85'
        _ana2color = '#ececec' # Closest X11 color: 'gray92'

        top.geometry("600x450+328+68")
        top.title("New Toplevel")
        top.configure(background="#d9d9d9")

        self.Label1 = tk.Label(top)
        self.Label1.place(relx=0.067, rely=0.644, height=51, width=94)
        self.Label1.configure(background="#d9d9d9")
        self.Label1.configure(disabledforeground="#a3a3a3")
        self.Label1.configure(foreground="#000000")
        self.Label1.configure(text='''Temperature''')

        self.Label2 = tk.Label(top)
        self.Label2.place(relx=0.367, rely=0.067, height=51, width=124)
        self.Label2.configure(background="#d9d9d9")
        self.Label2.configure(disabledforeground="#a3a3a3")
        self.Label2.configure(foreground="#000000")
        self.Label2.configure(text='''Robin''')

        self.Frame1 = tk.Frame(top)
        self.Frame1.place(relx=0.067, rely=0.422, relheight=0.167
                , relwidth=0.208)
        self.Frame1.configure(relief='groove')
        self.Frame1.configure(borderwidth="2")
        self.Frame1.configure(relief="groove")
        self.Frame1.configure(background="#d9d9d9")

        self.Frame2 = tk.Frame(top)
        self.Frame2.place(relx=0.583, rely=0.422, relheight=0.167
                , relwidth=0.208)
        self.Frame2.configure(relief='groove')
        self.Frame2.configure(borderwidth="2")
        self.Frame2.configure(relief="groove")
        self.Frame2.configure(background="#d9d9d9")

if __name__ == '__main__':
    vp_start_gui()







mss.py
#MSS 2021 | Alpha V0.2.1 - Status: Working
# Copyright Christian Jensen @covxx
# Contact cmjensenx@gmail.com
import subprocess
import math
import threading
import sys
import select
import os
import os.path
from pathlib import Path
from configparser import ConfigParser
config_object = ConfigParser()
from sense_hat import SenseHat
sense = SenseHat()
sense.clear()
sense.clear(255, 0, 0)
setf = Path("config.ini") #Path For Config File
arun = 'F' #var for first time check, gets over written by ftrun()
def clear_S(): #Call back for screen clearing
    os.system('cls' if os.name == 'nt' else 'clear') #Cross Platofrm scren clear
def ftrun(): #First time run check
    setf = Path("config.ini")
    if setf.is_file(): #checks if file exists
        global arun #Set arun to global status
        arun = "True"
        #print(arun) #Debug for first time check
        main_Menu() #Runs main menu
    else:
        ftsetup() #Runs first time setup
def ftsetup(): #First time setup config //In Progress
    clear_S()
    f= open('config.ini',"w+") #Creates config file
    global arun #Set arun to global status, this is done twice to make sure the var gets updated
    arun = "True"
    print(arun)
    main_Menu()
def main_Menu():
    clear_S()
    sense.clear()
    sense.clear(0, 255, 0)
    #print(arun) #Debug for first time check
    print ("Welcome to Robin Farming Intrustment")
    print ("----------------------------------------------")
    print ("1. Start Data Logging") #Run data logging session
    print ("2. Sensor Data") #Sub menu with sensor specific test
    print ("3. Settings") #Program settings
    print ("4. Shut Down") #Close Program
    print ("----------------------------------------------")
    mmi = int(input("Enter number selection to proceed:  "))
    if mmi == 1:
        print ("Loading...")
        start_dls() #| In Progress...
    elif mmi == 2:
        print ("Loading...")
        sm()
    elif mmi == 3:
        print ("Loading...")
        setng()
    elif mmi == 4: #May need to create a end prog function to close any open data filesd
        clear_S()
        print ("Shuting down application")
        exit() #Closes program
    else:
        main_Menu()
def sm(): #Menu Option Two
    clear_S()
    sense.clear(0, 255, 0)
    print ("Manual Data Logging Menu") #Prints infomation without saving data to file\server
    print ("1. Temperature") #Prints Temp
    print ("2. Humidity") #Print Humidity
    print ("3. Pressure") #Prints Pressure
    print ("4. Main Menu") #Back to main menu
    print ("----------------------------------------------")
    smi = int(input("Enter number selection to proceed:  "))
    if smi == 1:
        print ("Loading...")
        g_tp() #In progress
    elif smi == 2:
        print ("Loading...")
        g_hum()
    elif smi == 3:
        print ("Loading...")
        g_pbar()
    elif smi == 4:
        print("Loading main menu..")
        main_Menu()
    else: #Any other number re-runs menu
        sm()
        main_Menu()
def setng():
        clear_S()
        sense.clear(0, 255, 0)
        print ("Settings Menu")
        print ("1. Device Config") #Config options
        print ("2. Data Logging Timer") #Change Data logging end timer
        print ("3. Testing") #Testing menu
        print ("4. Main Menu") #Back to main menu
        print ("----------------------------------------------")
        seti = int(input("Enter number selection to proceed:  "))
        if seti == 1:
            print ("Loading...")
            #dcon() #Not implemented //Device config
        elif seti == 2:
            print ("Loading...")
            #dlet() #Not implemented //Data Logging end timer
        elif seti == 3:
            print ("Loading...")
            #pbar() #Not implemented //Testing menu
        elif seti == 4:
            print("Loading main menu..")
            main_Menu()
        else: #Any other number re-runs menu
            setng()
def g_pbar():
    clear_S()
    sense.clear()
    mdls_pbar = round(sense.get_pressure()) #rounds pressure reading
    print("Current Zone pressure Level:", mdls_pbar,"milliebars") #Sensehat unit is milliebars, future will convert
    input("Press any key to return to menu") #place holder for testing
    main_Menu()
def g_hum():
    clear_S()
    sense.clear()
    mdls_hum = sense.get_humidity()
    print("Current Zone Humidity Level:", mdls_hum,"%")
    input("Press any key to return to menu") #place holder for testing
    main_Menu()
def g_tp():
    clear_S()
    sense.clear()
    sense.clear(0, 0, 255)
    mdls_temp = round(sense.get_temperature()*9/5+32)
    print("Current Zone temperature:", mdls_temp,"F") #This will need to be looped with polling of every 3seconds
    input("Press any key to return to menu")
    main_Menu() #loop back to mm for testing, will be changed after loop
def start_dls(): #Data Logging start, loads config file to start //In Progress
#Config file needs to have user set parm to log (temp,Humidity,pressure)
#Needs to print current data
    logtim = 10 #Variable for log interval, will be set by settings menu and be a global var
    clear_S()
    sense.clear()
    dls_temp = round(sense.get_temperature()*9/5+32)
    print ("Starting data logging session...")
    log_end = time.time() + logtim 
    while time.time() < t_end:
        print("Current Zone temperature:", dls_temp,"F")
ftrun() #Check for first time setup - start program workflow


gui.tcl
#############################################################################
# Generated by PAGE version 4.25.1
#  in conjunction with Tcl version 8.6
#  Sep 24, 2019 08:50:00 PM EDT  platform: Windows NT
set vTcl(timestamp) ""


if {!$vTcl(borrow) && !$vTcl(template)} {

set vTcl(actual_gui_bg) #d9d9d9
set vTcl(actual_gui_fg) #000000
set vTcl(actual_gui_analog) #ececec
set vTcl(actual_gui_menu_analog) #ececec
set vTcl(actual_gui_menu_bg) #d9d9d9
set vTcl(actual_gui_menu_fg) #000000
set vTcl(complement_color) #d9d9d9
set vTcl(analog_color_p) #d9d9d9
set vTcl(analog_color_m) #ececec
set vTcl(active_fg) #000000
set vTcl(actual_gui_menu_active_bg)  #ececec
set vTcl(active_menu_fg) #000000
}




proc vTclWindow.top42 {base} {
    global vTcl
    if {$base == ""} {
        set base .top42
    }
    if {[winfo exists $base]} {
        wm deiconify $base; return
    }
    set top $base
    ###################
    # CREATING WIDGETS
    ###################
    vTcl::widgets::core::toplevel::createCmd $top -class Toplevel \
        -background $vTcl(actual_gui_bg) 
    wm focusmodel $top passive
    wm geometry $top 600x450+328+68
    update
    # set in toplevel.wgt.
    global vTcl
    global img_list
    set vTcl(save,dflt,origin) 0
    wm maxsize $top 1370 749
    wm minsize $top 120 1
    wm overrideredirect $top 0
    wm resizable $top 1 1
    wm deiconify $top
    wm title $top "New Toplevel"
    vTcl:DefineAlias "$top" "Toplevel1" vTcl:Toplevel:WidgetProc "" 1
    label $top.lab44 \
        -background $vTcl(actual_gui_bg) -disabledforeground #a3a3a3 \
        -font TkDefaultFont -foreground $vTcl(actual_gui_fg) \
        -text Temperature 
    vTcl:DefineAlias "$top.lab44" "Label1" vTcl:WidgetProc "Toplevel1" 1
    label $top.lab45 \
        -background $vTcl(actual_gui_bg) -disabledforeground #a3a3a3 \
        -font TkDefaultFont -foreground $vTcl(actual_gui_fg) -text Robin \
        -width 250 
    vTcl:DefineAlias "$top.lab45" "Label2" vTcl:WidgetProc "Toplevel1" 1
    frame $top.fra46 \
        -borderwidth 2 -relief groove -background $vTcl(actual_gui_bg) \
        -height 75 -width 125 
    vTcl:DefineAlias "$top.fra46" "Frame1" vTcl:WidgetProc "Toplevel1" 1
    frame $top.fra47 \
        -borderwidth 2 -relief groove -background $vTcl(actual_gui_bg) \
        -height 75 -width 125 
    vTcl:DefineAlias "$top.fra47" "Frame2" vTcl:WidgetProc "Toplevel1" 1
    ###################
    # SETTING GEOMETRY
    ###################
    place $top.lab44 \
        -in $top -x 40 -y 290 -width 94 -relwidth 0 -height 51 -relheight 0 \
        -anchor nw -bordermode ignore 
    place $top.lab45 \
        -in $top -x 220 -y 30 -width 124 -relwidth 0 -height 51 -relheight 0 \
        -anchor nw -bordermode ignore 
    place $top.fra46 \
        -in $top -x 40 -y 190 -anchor nw -bordermode ignore 
    place $top.fra47 \
        -in $top -x 350 -y 190 -anchor nw -bordermode ignore 

    vTcl:FireEvent $base <<Ready>>
}

set btop ""
if {$vTcl(borrow)} {
    set btop .bor[expr int([expr rand() * 100])]
    while {[lsearch $btop $vTcl(tops)] != -1} {
        set btop .bor[expr int([expr rand() * 100])]
    }
}
set vTcl(btop) $btop
Window show .
Window show .top42 $btop
if {$vTcl(borrow)} {
    $btop configure -background plum
}



start.py
#Robin 0.3 Update 
# Copyright Christian Jensen @covxx
# Contact cmjensenx@gmail.com
import sys
from sense_hat import SenseHat
import datetime
import glob
import requests
import os.path
from os import path
Configuring sensehat
sense = SenseHat()
sense.clear() #clears data from sensehat
temp = round(sense.get_temperature()*9/5+32)
humi = round(sense.get_humidity())
mbar = round(sense.get_pressure())
#Main section starts now
def frun(): #Checks if data file has already been created  and configures if it does not
    if os.path.isfile('data.txt'):  #File
        print("Loading Program...")
        main_Menu() #Calls menu
    else:
        print("Configuring Robin software, please wait...") #Configures settings
        tf = open('data.txt', 'w+') #Creates data file
        main_Menu()

def get_temp(): #Prints temp and is EOP
    print ("The temperature is currently:" ,temp, "F")
def hum(): #Humidity
    print ("The humidity is currently:",humi, "%")

def pbar(): #Pressure
    print ("The pressure is currently:",mbar,"mbar")

def main_Menu():
    print ("Welcome to Robin Farming Intrustment")
    print ("Ver. 0.3 - 1/20")
   # print ("The time currently is:"), ctime.strftime("%Y-%M-%D %H:%m:s") #Says current time; WIP/ May remove
    print ("----------------------------------------------")
    print("1. Start Data Logging") #Run data logging session
    print ("2. Sensor Data") #Sub menu with sensor specific test
    print ("3. Settings") #Program settings
    print ("4. Shut Down") #Close Program
    print ("----------------------------------------------")
    # MMI = Main Menu Input // Get temp or quit
    mmi = int(input("Enter number selection to proceed:  "))
    #print(mmi) # Test EOF
    if mmi == 1:
        print ("Loading...")
        start_mss()
    elif mmi == 2:
        print ("Loading...")
        sub_m()
    elif mmi == 3:
        print ("Loading...")
        settings()
    elif mmi == 4:
        print ("Logging out...")
        #shtdwn() | Shutdown class is currently not implmented
        sys.exit()
def sm():
    print ("Manual Data Logging Menu,")
    print ("1. Temperature") #Prints Temp
    print ("2. Humidity") #Print Humidity
    print ("3. Pressure") #Prints Pressure
    print ("----------------------------------------------")
    smi = int(input("Enter number selection to proceed:  "))
    if smi == 1:
        print ("Loading...")
        get_temp()
    elif smi == 2:
        print ("Loading...")
        hum()
    elif smi == 3:
        print ("Loading...")
        pbar()
def settings(): #Settings menu, but no settings
    print("Nothings here")
    main_menu() #Goes back to main menu until theres funticans

def shtdwn(): #Log out process to close data logging and save to local FS and network
    exit()

frun() #Program starts 
 


start_21.py
#MSS 2021 | Alpha V0.2 - Status: Working
# Copyright Christian Jensen @covxx
# Contact cmjensenx@gmail.com
import subprocess
import math
import threading
import sys
import select
import os
import os.path
from pathlib import Path
from configparser import ConfigParser
config_object = ConfigParser()
from sense_hat import SenseHat
sense = SenseHat()
sense.clear()
sense.clear(255, 0, 0)
pbar = round(sense.get_pressure())
tmp = round(sense.get_temperature())
humD = round(sense.get_humidity())
setf = Path("config.ini") #Path For Config File
arun = 'F' #var for first time check, gets over written by ftrun()
def clear_S(): #Call back for screen clearing
    os.system('cls' if os.name == 'nt' else 'clear') #Cross Platofrm scren clear
def ftrun(): #First time run check
    setf = Path("config.ini")
    if setf.is_file(): #checks if file exists
        global arun #Set arun to global status
        arun = "True"
        #print(arun) #Debug for first time check
        main_Menu() #Runs main menu
    else:
        ftsetup() #Runs first time setup
def ftsetup(): #First time setup config //In Progress
    clear_S()
    f= open('config.ini',"w+") #Creates config file
    global arun #Set arun to global status, this is done twice to make sure the var gets updated
    arun = "True"
    print(arun)
    main_Menu()
def main_Menu():
    clear_S()
    sense.clear()
    sense.clear(0, 255, 0)
    #print(arun) #Debug for first time check
    print ("Welcome to Robin Farming Intrustment")
    print ("----------------------------------------------")
    print("1. Start Data Logging") #Run data logging session
    print ("2. Sensor Data") #Sub menu with sensor specific test
    print ("3. Settings") #Program settings
    print ("4. Shut Down") #Close Program
    print ("----------------------------------------------")
    mmi = int(input("Enter number selection to proceed:  "))
    if mmi == 1:
        print ("Loading...")
        start_dls() #| In Progress...
    elif mmi == 2:
        print ("Loading...")
        sm()
    elif mmi == 3:
        print ("Loading...")
        setng()
    elif mmi == 4: #May need to create a end prog function to close any open data filesd
        clear_S()
        print ("Shuting down application")
        exit() #Closes program
    else:
        main_Menu()
def sm(): #Menu Option Two
    clear_S()
    sense.clear(0, 255, 0)
    print ("Manual Data Logging Menu") #Prints infomation without saving data to file\server
    print ("1. Temperature") #Prints Temp
    print ("2. Humidity") #Print Humidity
    print ("3. Pressure") #Prints Pressure
    print ("4. Main Menu") #Back to main menu
    print ("----------------------------------------------")
    smi = int(input("Enter number selection to proceed:  "))
    if smi == 1:
        print ("Loading...")
        g_tp() #In progress
    elif smi == 2:
        print ("Loading...")
        g_hum()
    elif smi == 3:
        print ("Loading...")
        g_pbar()
    elif smi == 4:
        print("Loading main menu..")
        main_Menu()
    else: #Any other number re-runs menu
        sm()
        main_Menu()
def setng():
        clear_S()
        sense.clear(0, 255, 0)
        print ("Settings Menu")
        print ("1. Device Config") #Config options
        print ("2. Data Logging Timer") #Change Data logging end timer
        print ("3. Testing") #Testing menu
        print ("4. Main Menu") #Back to main menu
        print ("----------------------------------------------")
        seti = int(input("Enter number selection to proceed:  "))
        if seti == 1:
            print ("Loading...")
            #dcon() #Not implemented //Device config
        elif seti == 2:
            print ("Loading...")
            #dlet() #Not implemented //Data Logging end timer
        elif seti == 3:
            print ("Loading...")
            #pbar() #Not implemented //Testing menu
        elif seti == 4:
            print("Loading main menu..")
            main_Menu()
        else: #Any other number re-runs menu
            setng()
def g_pbar():
    clear_S()
    sense.clear()
    mdls_pbar = round(sense.get_pressure()) #rounds pressure reading
    print("Current Zone pressure Level:", mdls_pbar,"milliebars") #Sensehat unit is milliebars, future will convert
    input("Press any key to return to menu") #place holder for testing
    main_Menu()
def g_hum():
    clear_S()
    sense.clear()
    mdls_hum = sense.get_humidity()
    print("Current Zone Humidity Level:", mdls_hum,"%")
    input("Press any key to return to menu") #place holder for testing
    main_Menu()
def g_tp():
    clear_S()
    sense.clear()
    sense.clear(0, 0, 255)
    mdls_temp = round(sense.get_temperature()*9/5+32)
    print("Current Zone temperature:", mdls_temp,"F") #This will need to be looped with polling of every 3seconds
    input("Press any key to return to menu")
    main_Menu() #loop back to mm for testing, will be changed after loop
def start_dls(): #Data Logging start, loads config file to start //In Progress
#Config file needs to have user set parm to log (temp,Humidity,pressure)
#Needs to print current data
    clear_S()
    sense.clear()
    dls_temp = round(sense.get_temperature()*9/5+32)
    print ("Starting data logging session...")
    #while true:
    print("Current Zone temperature:", dls_temp,"F") #This will need to be looped with polling of every 3seconds
ftrun() #Check for first time setup - start program workflow
